cdap-proto/src/main/java/co/cask/cdap/proto/TableInfo.java

```diff
@@ -90,7 +90,7 @@ public TableInfo(String tableName, String dbName, String owner, long creationTim
     this.lastAccessTime = lastAccessTime;
     this.retention = retention;
     this.partitionKeys = partitionKeys;
-    this.parameters = ImmutableMap.copyOf(parameters);
+    this.parameters = Collections.unmodifiableMap(parameters);
```

Comment:
I'm not sure if it matters - but this isn't exactly the same.
With the original code, changes to the input `parameters` would not be reflected in the unmodifiable view.